src/Schema/JsonApi/DynamicEntitySchema.php

```diff
@@ -0,0 +1,231 @@
+        if (!isset($this->resourceType)) {
+            $this->resourceType = strtolower(Inflector::pluralize($entityName));
+        }
+
+        parent::__construct($factory);
+    }
+
+    /**
+     * Get resource id.
+     *
+     * @param \Cake\ORM\Entity $entity Entity
+     * @return string
+     */
+    public function getId($entity)
+    {
+        return (string)$entity->get($this->idField);
+    }
+
+    /**
+     * NeoMerx override used to pass entity root properties to be shown
+     * as JsonApi `attributes`.
+     *
+     * @param \Cake\ORM\Entity $entity Entity
+     * @return array
+     */
+    public function getAttributes($entity)
+    {
+        if ($entity->has($this->idField)) {
+            $hidden = array_merge($entity->hiddenProperties(), [$this->idField]);
+            $entity->hiddenProperties($hidden);
+        }
+
+        $attributes = $entity->toArray();
+
+        // remove associated data so it won't appear inside jsonapi `attributes`
+        foreach ($this->_view->viewVars['_associations'] as $association) {
+            $associationKey = Inflector::tableize($association->table());
+
+            if (get_class($association) === 'Cake\ORM\Association\BelongsTo') {
+                $associationKey = Inflector::singularize($associationKey);
+            }
+
+            unset($attributes[$associationKey]);
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * NeoMerx override used to pass associated entity names to be used for
+     * generating JsonApi `relationships`.
+     *
+     * JSON API optional `related` links not implemented yet.
+     *
+     * @param \Cake\ORM\Entity $resource Entity object
+     * @param bool $isPrimary True to add resource to data section instead of included
+     * @param array $includeRelationships Used to fine tune relationships
+     * @return array
+     */
+    public function getRelationships($resource, $isPrimary, array $includeRelationships)
+    {
+        $relations = [];
+
+        foreach ($this->_view->viewVars['_associations'] as $association) {
+            $associationKey = Inflector::tableize($association->name());
+
+            if (get_class($association) === 'Cake\ORM\Association\BelongsTo') {
+                $associationKey = Inflector::singularize($associationKey);
+            }
+
+            $data = $resource->$associationKey;
+            if (!$data) {
+                continue;
+            }
+
+            $relations[$associationKey] = [
+                self::DATA => $data,
+                self::SHOW_SELF => true,
+                self::SHOW_RELATED => false,
+            ];
+        }
+
+        return $relations;
+    }
+
+    /**
+     * NeoMerx override to generate Cake belongsTo and hasMany links.
+     *
+     * @param \Cake\ORM\Entity $resource Entity
+     * @param string $name Relationship name in lowercase singular or plural
+     * @param array $meta Optional array with meta information
```

Comment:
Just removed the fqcn, thanks for pointing it out. I am pausing the other return type changes until PR #416 settles.